js/views/modals/orderDetail/contractTab/ContractTab.js

```diff
@@ -53,19 +53,33 @@ export default class extends BaseVw {
   }
 
   renderStatus() {
+    const iconBaseClass = 'margRSm flexNoShrink';
     let msg = '';
 
     if (this.isCase) {
       // Cut a corner with some html embedded here. If the html get more elaborate than this,
       // we should probably break this out into its own template.
       if (this.model.bothContractsValid) {
-        msg = '<p class="clrTEm flexVCent"><span class="ion-ios-checkmark-outline margRSm">' +
-          `</span>${app.polyglot.t('orderDetail.contractTab.bothContractsValid')}</p>`;
+        const icon = `<span class="${iconBaseClass} tx1 ion-ios-checkmark-outline"></span>`;
+        const msgText = !this.model.vendorProcessingError ?
+          app.polyglot.t('orderDetail.contractTab.bothContractsValid') :
+          app.polyglot.t('orderDetail.contractTab.validBuyerVendorProcessingError');
+        const msgHtml = `<span>${msgText}</span>`;
+        msg = `<p class="clrTEm flexVCent">${icon}${msgHtml}</p>`;
       } else if (!this.model.get('vendorContract')) {
-        msg = '<p class="flexVCent"><span class="ion-android-warning margRSm clrTAlert">' +
-          `</span>${app.polyglot.t('orderDetail.contractTab.vendorContractNotArrived')}</p>`;
+        const icon = `<span class="${iconBaseClass} clrTAlert ion-android-warning"></span>`;
+        const processingErrorKey =
+          'orderDetail.contractTab.vendorContractNotArrivedPotentialProcErr';
+        const buyerContract = this.model.get('buyerContract');
+        const buyerShowsVendorProcErr =
+          buyerContract && Array.isArray(buyerContract.get('errors'));
```

Comment:
Does the array check here need to check length, or is an empty array not possible?

Reply:
Same as above, based on what the current server behavior is (and CP confirmed that we should be able to rely on this behavior) the error field is never added to the contract until the point the vendor has an error processing the order. At that point, the Array will be added in and it will remain in the contract, which is important because it allows us to know whether there was a processing error even after the state moves on.

I don't know if there will ever be a situation where there is an empty array, but if there is that would be considered a processing error without a reason, which the client handles with generic messaging.